source/module_hamilt/ks_pw/veff_pw.cpp

```diff
@@ -46,38 +46,47 @@ void Veff<OperatorPW>::act
     {
         if (this->npol == 1)
         {
-            const double* current_veff = &(this->veff[0](current_spin, 0));
             wfcpw->recip2real(tmpsi_in, porter, ik);
-            for (int ir = 0; ir < this->veff->nc; ++ir)
+            // NOTICE: when MPI threads are larger than number of Z grids
+            // veff would contain nothing, and nothing should be do in real space
+            // but the 3DFFT can not be skipped, it will cause stuck
```

Comment:
```suggestion
            // veff would contain nothing, and nothing should be done in real space
            // but the 3DFFT can not be skipped, it will cause hanging
```